### PR TITLE
Enable gRPC when downstream is HTTP/1.1

### PR DESCRIPTION
**Pull request by qiwzhang**
Signed-off-by: Wayne Zhang <qiwzhang@google.com>

In order to support gRPC when downstream is HTTP/1.1,   
1) Envoy Http_connection_manager Http1ProtocolOptions need to set enable_trailers=true
2) Need to add grpc_metadata_scrubber to remove Content-Length for gRPC reqeuests.

Add a configMgr option and flag EnableGrpcForHttp1.  
Default is on, since it doesn't harm anything.

### CODE REVIEW

**Comment by qiwzhang** [comment]
PTAL

**Review by nareddyt — APPROVED** [review]

**Review by toddbeckman-g — APPROVED** [review]

File: src/go/configmanager/flags/flags.go [hunk 1]
```diff
@@ -160,6 +162,7 @@ func EnvoyConfigOptionsFromFlags() options.ConfigGeneratorOptions {
 		SuppressEnvoyHeaders:                    *SuppressEnvoyHeaders,
 		UnderscoresInHeaders:                    *UnderscoresInHeaders,
 		ServiceControlNetworkFailOpen:           *ServiceControlNetworkFailOpen,
+		EnableGrpcForHttp1:                      *EnableGrpcForHttp1,
```
Comment (JLXIA) [inline src/go/configmanager/flags/flags.go:165]: Do we need this flag here, or just default to false? I thought this is only needed for ApiGateway when it runs on Titanium.

Comment (TAOXUY) [reply src/go/configmanager/flags/flags.go:165]: +1.

Comment (qiwzhang) [reply src/go/configmanager/flags/flags.go:165]: Decided to add a Option to ConfigMgr.  We don't need a flag.  But for now,  every ConfigMgr option has a flag, so just add a flag.

**Pull request merged**